File: src/main/java/com/m2t/hachikodictionary/model/Confirmation.java
package com.m2t.hachikodictionary.model;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document("confirmations")
public class Confirmation {

    @Id
    private String id;

    private String email;
    private String token;

    public Confirmation(String email, String token) {
        this.email = email;
        this.token = token;
    }
}

File: src/main/java/com/m2t/hachikodictionary/model/Confirmation.kt
package com.m2t.hachikodictionary.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.GenericGenerator

@Entity
@Table(name = "confirmations")
data class Confirmation(
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator") // Generate UUID
    val id: String,

    @Column(unique = true)
    val email: String,
    val token: String,
) {
    constructor(email: String, token: String) : this(
        "",
        email,
        token
    )
}